python/exercise2/time_window.py

```diff
@@ -20,12 +20,20 @@ def __init__(self, start_time, end_time, delta):
                 
         """
         
+        # Error checking of delta input
         if delta > 23:
           raise ValueError("invalid delta {} should be [0-23] hours".format(delta))
-        
+
+        if delta < 0:
+          raise ValueError("invalid delta {} should be [0-23] hours".format(delta))
+
+        if delta == 0:
+          raise ValueError("invalid delta {} cannot be exactly 0 hours".format(delta))
```

Comment:
```suggestion
        if delta <= 0:
          raise ValueError("invalid delta {} should be [1-23] hours".format(delta))
```
we can combine these two conditions into one.